src/js/github.js

```diff
@@ -0,0 +1,123 @@
+github = {};
+
+// The client ID of a GitHub OAuth app registered at https://github.com/settings/developers.
+// The “callback URL” of that app points to https://www.puzzlescript.net/auth.html.
+// If you’re running from another host name, sharing might not work.
+OAUTH_CLIENT_ID = "211570277eb588cddf44";
+
+github.authURL = function() {
+	// Generates 32 letters of random data, like "liVsr/e+luK9tC02fUob75zEKaL4VpQn".
+	var randomState = window.btoa(Array.prototype.map.call(
+		window.crypto.getRandomValues(new Uint8Array(24)),
+		function(x) { return String.fromCharCode(x); }).join(""));
+
+	return "https://github.com/login/oauth/authorize"
+		+ "?client_id=" + OAUTH_CLIENT_ID
+		+ "&scope=gist"
+		+ "&state=" + randomState
+		+ "&allow_signup=true";
+};
+
+github.signOut = function() {
+	storage_remove("oauth_access_token");
+};
+
+github.isSignedIn = function() {
+	var token = storage_get("oauth_access_token");
```

Comment:
nit: Maybe worth extracting a `getToken()` helper?

We call `storage_get("oauth_access_token")` a few times, which might be simpler to maintain if handled in one place.

Reply:
nah it's ok